sh.c

```diff
@@ -2043,6 +2043,9 @@ void mark_mutable_variables_statement(ast node) {
   } else if (op == WHILE_KW) {
     mark_mutable_variables_statement(get_child(node, 0));
     if (get_child(node, 1)) mark_mutable_variables_body(get_child(node, 1));
+  } else if (op == DO_KW) {
+    mark_mutable_variables_statement(get_child(node, 0));
+    if (get_child(node, 1)) mark_mutable_variables_body(get_child(node, 1));
```

Comment:
I think in this case the child 0 is optional and child 1 is non-optional, so the opposite of the `WHILE_KW` case above. See https://github.com/udem-dlteam/pnut/blob/master/pnut.c#L3020-L3034

Reply:
Yep, wasn't sure of the structure, meant to ask you. Working on the `exe` backend right now.